monai/transforms/intensity/array.py

```diff
@@ -1318,7 +1318,7 @@ def __call__(self, img: Union[np.ndarray, torch.Tensor]) -> Union[torch.Tensor,
         # map back
         img = self.inv_shift_fourier(k, n_dims)
 
-        return img if self.as_tensor_output else img.cpu().detach().numpy()
+        return img if self.as_tensor_output else img.cpu().detach().numpy()  # type: ignore
```

Comment:
May I know why adding type-hints ignore here?

Reply:
`monai.utils.deprecated_arg` as a decorator couldn't preserve the original type annotations of the function to be wrapped...I spend some time on this but couldn't find a good solution yet